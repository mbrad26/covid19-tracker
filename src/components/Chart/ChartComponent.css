div #chart {
  /* height: 350px; */
}

.chartjs-render-monitor {
  /* margin-bottom: 0px; */
}
