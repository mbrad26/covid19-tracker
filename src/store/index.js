import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers';
import rootSaga from '../sagas';

const logger = createLogger();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  applyMiddleware(saga, logger)
);

// console.log('STORE: ', store.getState());

saga.run(rootSaga);

export default store;
