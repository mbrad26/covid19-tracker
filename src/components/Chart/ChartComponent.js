import React from 'react';
import { Line } from 'react-chartjs-2';

import './ChartComponent.css';

const ChartComponent = () => (
  <div className='shadow' id='chart'>
    Chart
    <Line />
  </div>
)

export default ChartComponent;
