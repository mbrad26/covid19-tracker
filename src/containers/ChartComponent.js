import { connect } from 'react-redux';
import { getData } from '../selectors/data';

import ChartComponent from '../components/Chart/ChartComponent';
import { doHistoricalDataLoading } from '../actions/historicalData.js';

const mapStateToProps = ({ historicalDataState }) => ({
  data: getData(historicalDataState),
});

const mapDispatchToProps = dispatch => ({
  loadingData: () => dispatch(doHistoricalDataLoading()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChartComponent);
