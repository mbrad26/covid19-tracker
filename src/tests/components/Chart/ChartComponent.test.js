import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import store from '../../../store';
import ChartComponent from '../../../components/Chart/ChartComponent';

describe('ChartComponent', () => {
  it('renders snapshot', () => {
    const { container } = render(
      <ChartComponent />
      );

    expect(container.firstChild).toMatchSnapshot();
  });
});
