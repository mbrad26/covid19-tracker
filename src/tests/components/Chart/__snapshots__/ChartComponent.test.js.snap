// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartComponent renders snapshot 1`] = `
<canvas
  class="chartjs-render-monitor"
  height="0"
  style="display: block;"
  width="0"
/>
`;
